### PR TITLE
PP-7683 apps into test ECR

### PR DESCRIPTION
**Pull request by katstevens**
TODO:
- ~we have duplicated the ECR registry resource for each repo, this could probably be DRYer~
- ~ci/pipelines/toolbox-test-ecr.yml could be renamed but we'll do that later for ease of reviewing just now.~
- ~revert https://github.com/alphagov/pay-ci/pull/363/commits/9e837183aa45cbbe9f25de92d6c7f47dc6e26be0 to put the meta job branch back to `master`~

Used the group formatting to combine all the app tasks into one pipeline, as per the pr-ci pipeline.

Pulling the pay-cardid repo is a special case in that it has a private submodule. To get the repo we have ignored the submodules when pulling:

```
      - get: cardid-git-release
        trigger: true
        params:
          submodules: none
```
And then updated the repo in a separate task akin to https://github.com/alphagov/pay-ci/blob/master/ci/pipelines/pr.yml#L1253-L1286.

### CODE REVIEW

File: ci/pipelines/apps-test-ecr.yml [hunk 1]
```diff
@@ -1,34 +1,168 @@
+aws_config: &aws_config
```
Comment (dj-maisy) [inline ci/pipelines/apps-test-ecr.yml:1]: I assume this is fine to do outside of a definition? This is fine if it is. 👍

File: ci/pipelines/apps-test-ecr.yml [hunk 2]
```diff
@@ -114,42 +276,214 @@ jobs:
...
-            - |
-              COMMIT_SHA=$(cat git-toolbox-pre-release/.git/HEAD)
-              GIT_TAG=$(cat git-toolbox-pre-release/.git/ref)
-              echo "$COMMIT_SHA $GIT_TAG ready-staging" > docker_tags/docker_tags.txt
-          outputs:
-            - name: docker_tags
-      - put: ecr-registry-test
+              - -ec
+              - |
+                # cardid-data submodule's url is defined as ssh yet concourse
```
Comment (dj-maisy) [inline ci/pipelines/apps-test-ecr.yml:362]: Awesome description. 👍

File: ci/pipelines/apps-test-ecr.yml [hunk 3]
```diff
@@ -44,67 +178,95 @@ resource_types:
...
+      - publicauth-image-to-test-ecr
+  - name: selfservice
+    jobs:
+      - selfservice-image-to-test-ecr
+
+definitions:
+  - &pull-image-from-dockerhub
+    task: pull-image-from-dockerhub
+    privileged: true
+    file: toolbox-test-ecr/ci/tasks/pull-image-from-dockerhub.yml
```
Comment (dj-maisy) [inline ci/pipelines/apps-test-ecr.yml:223]: Also happy that this has been moved into a file. Gets rid of repetition. 😄

File: ci/pipelines/apps-test-ecr.yml (hunk 1 above)
Comment (katstevens) [reply ci/pipelines/apps-test-ecr.yml:1]: We're doing the same for `docker_credentials` below. It seems to work!

Comment (rorymalcolm) [reply ci/pipelines/apps-test-ecr.yml:1]: these are just pure YAML anchors - they're in the spec! https://yaml.org/spec/1.2/spec.html#id2765878

**Review by rorymalcolm — APPROVED** [review]
LGTM, good work team

**Pull request merged**